salt/modules/kubernetes.py

```diff
@@ -168,8 +169,9 @@ def _setup_conn(**kwargs):
         with tempfile.NamedTemporaryFile(prefix='salt-kube-', delete=False) as k:
             k.write(base64.b64decode(client_key))
             kubernetes.client.configuration.key_file = k.name
-    else:
-        kubernetes.client.configuration.key_file = None
```

Comment:
Why are you dropping this line as part of this PR?

Reply:
Did you see [this test](https://github.com/saltstack/salt/pull/44231/files#diff-ca4139941d80e779d3053f879b625cccR149)? This is the [original code](https://github.com/saltstack/salt/blob/405b88243857fa219029c6dd5555df3f7b17ed68/salt/modules/kubernetes.py#L165-L172) with logic that is roughly as follows:
 
```
if kubernetes.client-key-file or kwarg passed:
    set the kubernetes key_file configuration to the passed value
 
if kubernetes.client-key-data config option or kwarg passed:
    create a temp file with the key data and set the key_file to the value
else:
    zero out the key_file
```
 
The problem with that code, is that it entirely overwrites the `key_file` attribute of `kubernetes.client.configuration` which entirely invalidates the `kubernetes.client-key-file` config option. If you run that unit test with those two lines, the test will fail, which hopefully we can agree is incorrect. Also, the default of `kubernetes.client.configuration.key_file` is `None` so setting it doesn't seem to make any sense. Redundant code is redundant and all.

Without those lines, ssl key auth actually works using a file. This is [documented](https://github.com/saltstack/salt/pull/44231/commits/1c821c0e77de58892c77d8e55386fac25e518c31) in the commit message if that is ok with you.

Reply:
Also, it was impossible for me to authenticate salt with my tectonic cluster without this patch. It works.

Reply:
Sorry, but I think this is wrong. Setting this to None was made to be able to switch between multiple clusters.
This is a "reset to save defaults".

The bug is in line 167 which should be an "elif". This should prevent resetting key_file to None, when the keyfile was provided in kwargs.

Reply:
@mcalmer Yeah I considered doing an elif, but wanted some feedback, hence the cc. The test I wrote will cover either approach and prevent regressions, so I'll try to update this PR tonight. Thanks for the clarification. Were you planning on using this module to switch clusters via `salt` / `salt-call` kwargs, or some other magic? I ask because I'm using it to manage a CoreOS tectonic cluster and will be using it to manage a few other open sauce k8s clusters as well.

Reply:
I think switching via different pillar data is the best usecase for working with multiple clusters.